src/util/workers.c

```diff
@@ -105,44 +131,19 @@ void workersThreadPool_Destroy(void) {
   redisearch_thpool_destroy(_workers_thpool);
 }
 
-void workersThreadPool_Activate() {
-  if (USE_BURST_THREADS()) { /* Configure here anything that needs to know it can use the thread pool */
-    // Change VecSim write mode temporarily for fast RDB loading of vector index (if needed).
-    VecSim_SetWriteMode(VecSim_WriteAsync);
-
-    // Finally, log that we've enabled the thread pool.
-    RedisModule_Log(RSDummyContext, "notice", "Enabled workers threadpool of size %lu",
-                    RSGlobalConfig.numWorkerThreads);
-  }
+void workersThreadPool_OnEventStart() {
+  in_event = true;
+  workersThreadPool_SetNumWorkers();
 }
 
-void workersThreadPool_waitAndTerminate(RedisModuleCtx *ctx) {
-    // Wait until all the threads are finished the jobs currently in the queue. Note that we call
-    // block main thread while we wait, so we have to make sure that number of jobs isn't too large.
-    if (RSGlobalConfig.numWorkerThreads == 0) return;
+void workersThreadPool_OnEventEnd(bool wait) {
+  in_event = false;
+  workersThreadPool_SetNumWorkers();
+  // Wait until all the threads are finished the jobs currently in the queue. Note that we call
+  // block main thread while we wait, so we have to make sure that number of jobs isn't too large.
+  // no-op if numWorkerThreads == minOperationWorkers == 0
```

Comment:
no-op means is doesn't have any affect if blahblah?

Reply:
yes